pkg/cluster/k8sres.go

```diff
@@ -1619,12 +1619,10 @@ func (c *Cluster) generateService(role PostgresRole, spec *acidv1.PostgresSpec)
 		}
 
 		c.logger.Debugf("final load balancer source ranges as seen in a service spec (not necessarily applied): %q", serviceSpec.LoadBalancerSourceRanges)
-		// spec.externalTrafficPolicy can define externalTrafficPolicy for the service
-		// if spec.externalTrafficPolicy is not defined default value is "Cluster"
-		if spec.ExternalTrafficPolicy != nil {
-			serviceSpec.ExternalTrafficPolicy = *spec.ExternalTrafficPolicy
+		if c.OpConfig.ExternalTrafficPolicy == "" {
```

Comment:
With the absence of any kind of error handling just dealing with the empty case does not make sense or? it is already set to default to "Cluster" if not set. so just line 1625 is good enough.

Reply:
I am not sure if I am doing anything wrong, but I don't get default values in the test.

Reply:
Yes but in no case is c.opconfig.externaltrafficpolicy "", which is fine. I would just argue this whole if statement can just be

```
serviceSpec.ExternalTrafficPolicy = v1.ServiceExternalTrafficPolicyType(c.OpConfig.ExternalTrafficPolicy)
```

You have the default properly set, so it should be cluster for everyone.

Everyone else switches it to Local.

And everyone else using something invalid gets an error.

So why deal/cover `"""` case extra?

Reply:
I mean, if I run tests and I don't define c.opConfig.ExternalTrafficPolicy, then I get "" here. Not the default "Cluster" that I expect.

Reply:
I guess it's just because the way how the config is defined in tests.